src/test/java/org/jitsi/meet/test/pageobjects/web/KnockingParticipantList.java

```diff
@@ -0,0 +1,177 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.meet.test.pageobjects.web;
+
+import org.jitsi.meet.test.util.*;
+import org.jitsi.meet.test.web.*;
+import org.openqa.selenium.*;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.interactions.*;
+
+import java.util.*;
+import java.util.logging.*;
+
+/**
+ * Representation of the list of knocking participants with allow and reject action.
+ * Appears to moderators when Lobby is enabled for a room.
+ */
+public class KnockingParticipantList
+{
+    /**
+     * The id of the participants list element.
+     */
+    private final static String KNOCKING_PARTICIPANT_LIST_ID = "knocking-participant-list";
+
+    /**
+     * The xpath to locate all knocking participants list.
+     */
+    private final static String PARTICIPANTS_XPATH = "//div[@id='" + KNOCKING_PARTICIPANT_LIST_ID + "']//ul//li";
+
+    /**
+     * TestIds for the participant details: avatar, name, email and the action buttons for allow and reject.
+     */
+    private final static String PARTICIPANT_AVATAR_TEST_ID = "knockingParticipant.avatar";
+    private final static String PARTICIPANT_EMAIL_TEST_ID = "knockingParticipant.email";
+    private final static String PARTICIPANT_NAME_TEST_ID = "knockingParticipant.name";
+    private final static String PARTICIPANT_ALLOW_TEST_ID = "lobby.allow";
+    private final static String PARTICIPANT_REJECT_TEST_ID = "lobby.reject";
+
+    /**
+     * The participant used to interact with the knocking participants list.
+     */
+    private final WebParticipant participant;
+
+    /**
+     * Initializes a new {@link KnockingParticipantList} instance.
+     *
+     * @param participant the participant for this {@link KnockingParticipantList}.
+     */
+    public KnockingParticipantList(WebParticipant participant)
+    {
+        this.participant = participant;
+    }
+
+    /**
+     * Returns the list of knocking participants. If the UI list is not found will return empty list of participants.
+     * @return list of knocking participants.
+     */
+    public List<Participant> getParticipants()
+    {
+        List<Participant> participants = new ArrayList<>();
+
+        try
+        {
+            TestUtils.waitForDisplayedElementByID(participant.getDriver(), KNOCKING_PARTICIPANT_LIST_ID, 3);
+        }
+        catch(TimeoutException ex)
+        {
+            // if the list is missing return empty list of participants
+            Logger.getGlobal().log(Level.WARNING, "No knocking participant list ");
```

Comment:
wasn't there a util method for logging somewhere?

Reply:
Yeah, I was thinking the same and did not find anything ...